test/e2e/runner/rpc.go

```diff
@@ -51,7 +59,23 @@ func waitForHeight(testnet *e2e.Testnet, height int64) (*types.Block, *types.Blo
 				maxResult = result
 				lastIncrease = time.Now()
 			}
+
 			if maxResult != nil && maxResult.Block.Height >= height {
+				// the node has achieved the target height!
+
+				// add this node to the set of target
+				// height nodes
+				nodesAtHeight[node.Name] = struct{}{}
+
+				// if not all of the nodes that we
+				// have clients for have reached the
+				// target height, keep trying.
+				if len(nodesAtHeight) < len(clients) {
```

Comment:
I don't think this will work because `clients` starts off as an empty map and gets filled gradually. Therefore, if the first node has reached the height already we will have len(clients) == 1 == len(nodesAtHeight) and thus we'll finish before checking in with any of the other nodes. We need to loop through all the nodes first and fill up the `clients` map and then start querying the heights

Reply:
good call.

Reply:
good call.